### DISCUSSION TITLE
Proposal: Accel-PPP APT (deb) repository

### DISCUSSION THREAD

**Discussion by svlobanov** [accel-ppp/accel-ppp#148] [Installation]
Currently, accel-ppp doesn't have any public APT repository. The idea is to package accel-ppp as three separate package: accel-ppp itself (userspace app), ipoe dkms driver, vlan_mon dkms driver, build .deb packages (using github actions), add them to apt repo and publish apt repo to github. 
Without DKMS, it is not realistic to build accel-ppp for every kernel version (moreover, kernel might be update from backport repo and so on)

DMKS support PR is here: https://github.com/accel-ppp/accel-ppp/pull/145
Accel-ppp APT repository  (dev version, do not use in production!) is here: https://github.com/svlobanov/accel-ppp-deb , build/upload code is here: https://github.com/svlobanov/accel-ppp-deb/blob/main/.github/workflows/build-and-upload.yaml

**Comment by andamasov** [comment]
@svlobanov, how about moving this issue to discussions and continue there?

**Comment by andamasov** [comment]
@svlobanov  Do we want our repo or package included in Ubuntu default repos?

**Reply by svlobanov** [reply]
I propose our repo. Other options are: ubuntu launchpad ppa (ubuntu only), debian/ubuntu default repositories. Both variants require preparing deb-src packages, but cmake can't do that, it will require to implement another way of building deb packages.

**Reply by andamasov** [reply]
I'm okay with our repo approach, but I think it would be great to have that in default repos in the long term.
I quickly googled it, and it seems that https://cmake.org/cmake/help/latest/module/CPack.html can generate sources. 
It's not a top priority, but we can investigate that path later.

**Reply by svlobanov** [reply]
cmake doesn't support building deb-src:
supported formats: https://cmake.org/cmake/help/latest/cpack_gen/archive.html#cpack_gen:CPack%20Archive%20Generator , https://cmake.org/cmake/help/latest/manual/cpack-generators.7.html#manual:cpack-generators(7)

debian documentation has such howto (how to build deb-src if project uses cmake): https://www.debian.org/doc/manuals/debmake-doc/ch08.en.html#cmake-multi

**Reply by andamasov** [reply]
Cmake itself no, but it can generate CPack config, and CPack can prepare a source archive  that can after be used with dh 
we can come back to that later if we decide to get into default distribution repositories

**Comment by andamasov** [comment]
This may be interesting https://github.com/rpatterson/github-apt-repos

**Reply by svlobanov** [reply]
I'll check this project, maybe it will simplify the solution

**Comment by OmlineEditor** [comment]
let's use the official repository to make it convenient to install
https://github.com/orgs/accel-ppp/discussions/197
